File: javascript/tennis/models/tennisService.js
var TennisService = function (player1Name, player2Name) {
  this.P1point = 0;
  this.P2point = 0;

  this.P1res = "";
  this.P2res = "";

  this.player1Name = player1Name;
  this.player2Name = player2Name;
};

TennisService.prototype.displayScore = function (p1S, p2S) {
  this.P1point = p1S;
  this.P2point = p2S;
  return this.getScore();
};

TennisService.prototype.getScore = function () {
  var score = "";

  if (this.P1point === this.P2point && this.P1point < 3) {
    if (this.P1point === 0)
      score = "love";
    if (this.P1point === 1)
      score = "fifteen";
    if (this.P1point === 2)
      score = "thirty";
    score += "-all";
  }
  if (this.P1point === this.P2point && this.P1point > 2)
    score = "deuce";

  if (this.P1point > 0 && this.P2point === 0) {
    if (this.P1point === 1)
      this.P1res = "fifteen";
    if (this.P1point === 2)
      this.P1res = "thirty";
    if (this.P1point === 3)
      this.P1res = "forty";

    this.P2res = "love";
    score = this.P1res + "-" + this.P2res;
  }
  if (this.P2point > 0 && this.P1point === 0) {
    if (this.P2point === 1)
      this.P2res = "fifteen";
    if (this.P2point === 2)
      this.P2res = "thirty";
    if (this.P2point === 3)
      this.P2res = "forty";

    this.P1res = "love";
    score = this.P1res + "-" + this.P2res;
  }

  if (this.P1point > this.P2point && this.P1point < 4) {
    if (this.P1point === 2)
      this.P1res = "thirty";
    if (this.P1point === 3)
      this.P1res = "forty";
    if (this.P2point === 1)
      this.P2res = "fifteen";
    if (this.P2point === 2)
      this.P2res = "thirty";
    score = this.P1res + "-" + this.P2res;
  }
  if (this.P2point > this.P1point && this.P2point < 4) {
    if (this.P2point === 2)
      this.P2res = "thirty";
    if (this.P2point === 3)
      this.P2res = "forty";
    if (this.P1point === 1)
      this.P1res = "fifteen";
    if (this.P1point === 2)
    this.P1res = "thirty";
    score = this.P1res + "-" + this.P2res;
  }

  if (this.P1point > this.P2point && this.P2point >= 3) {
    score = "advantage player1";
  }

  if (this.P2point > this.P1point && this.P1point >= 3) {
    score = "advantage player2";
  }

  if (this.P1point >= 4 && this.P2point >= 0 && (this.P1point - this.P2point) >= 2) {
    score = "win for player1";
  }
  if (this.P2point >= 4 && this.P1point >= 0 && (this.P2point - this.P1point) >= 2) {
    score = "win for player2";
  }
  return score;
};

TennisService.prototype.SetP1Score = function (number) {
  var i;
  for (i = 0; i < number; i++) {
    this.P1Score();
  }
};

TennisService.prototype.SetP2Score = function (number) {
  var i;
  for (i = 0; i < number; i++) {
    this.P2Score();
  }
};

TennisService.prototype.P1Score = function () {
  this.P1point++;
};

TennisService.prototype.P2Score = function () {
  this.P2point++;
};

TennisService.prototype.wonPoint = function (player) {
  if (player === "player1")
    this.P1Score();
  else
    this.P2Score();
};

module.exports = TennisService;

File: python/tennis/tennis_service.py
class TennisService:
    def __init__(self):
        self.p1points = 0
        self.p2points = 0
        self.player1Name = 'player1'
        self.player2Name = 'player2'

    def display_score(self, p1score, p2Score):
        self.p1points = p1score
        self.p2points = p2Score
        return self.score()

    def won_point(self, player_name):
        if player_name == self.player1Name:
            self.P1Score()
        else:
            self.P2Score()

    def score(self):
        result = ""
        if (self.p1points == self.p2points and self.p1points < 3):
            if (self.p1points==0):
                result = "love"
            if (self.p1points==1):
                result = "fifteen"
            if (self.p1points==2):
                result = "thirty"
            result += "-all"
        if (self.p1points==self.p2points and self.p1points>2):
            result = "deuce"

        P1res = ""
        P2res = ""
        if (self.p1points > 0 and self.p2points==0):
            if (self.p1points==1):
                P1res = "fifteen"
            if (self.p1points==2):
                P1res = "thirty"
            if (self.p1points==3):
                P1res = "forty"

            P2res = "love"
            result = P1res + "-" + P2res
        if (self.p2points > 0 and self.p1points==0):
            if (self.p2points==1):
                P2res = "fifteen"
            if (self.p2points==2):
                P2res = "thirty"
            if (self.p2points==3):
                P2res = "forty"

            P1res = "love"
            result = P1res + "-" + P2res


        if (self.p1points>self.p2points and self.p1points < 4):
            if (self.p1points==2):
                P1res="thirty"
            if (self.p1points==3):
                P1res="forty"
            if (self.p2points==1):
                P2res="fifteen"
            if (self.p2points==2):
                P2res="thirty"
            result = P1res + "-" + P2res
        if (self.p2points>self.p1points and self.p2points < 4):
            if (self.p2points==2):
                P2res="thirty"
            if (self.p2points==3):
                P2res="forty"
            if (self.p1points==1):
                P1res="fifteen"
            if (self.p1points==2):
                P1res="thirty"
            result = P1res + "-" + P2res

        if (self.p1points > self.p2points and self.p2points >= 3):
            result = "advantage " + self.player1Name

        if (self.p2points > self.p1points and self.p1points >= 3):
            result = "advantage " + self.player2Name

        if (self.p1points>=4 and self.p2points>=0 and (self.p1points-self.p2points)>=2):
            result = "win for " + self.player1Name
        if (self.p2points>=4 and self.p1points>=0 and (self.p2points-self.p1points)>=2):
            result = "win for " + self.player2Name
        return result

    def SetP1Score(self, number):
        for i in range(number):
            self.P1Score()

    def SetP2Score(self, number):
        for i in range(number):
            self.P2Score()

    def P1Score(self):
        self.p1points +=1


    def P2Score(self):
        self.p2points +=1


    def display_score_with_lang(self, language, p1score, p2Score):
        self.p1points = p1score
        self.p2points = p2Score
        return self.french_score()


    def french_score(self):
        result = ""
        if (self.p1points == self.p2points and self.p1points < 3):
            if (self.p1points==0):
                result = "zéro"
            if (self.p1points==1):
                result = "quinze"
            if (self.p1points==2):
                result = "trente"
            result += "-partout"
        if (self.p1points==self.p2points and self.p1points>2):
            result = "égalité"

        P1res = ""
        P2res = ""
        if (self.p1points > 0 and self.p2points==0):
            if (self.p1points==1):
                P1res = "quinze"
            if (self.p1points==2):
                P1res = "trente"
            if (self.p1points==3):
                P1res = "quarante"

            P2res = "zéro"
            result = P1res + "-" + P2res
        if (self.p2points > 0 and self.p1points==0):
            if (self.p2points==1):
                P2res = "quinze"
            if (self.p2points==2):
                P2res = "trente"
            if (self.p2points==3):
                P2res = "quarante"

            P1res = "zéro"
            result = P1res + "-" + P2res


        if (self.p1points>self.p2points and self.p1points < 4):
            if (self.p1points==2):
                P1res="trente"
            if (self.p1points==3):
                P1res="quarante"
            if (self.p2points==1):
                P2res="quinze"
            if (self.p2points==2):
                P2res="trente"
            result = P1res + "-" + P2res
        if (self.p2points>self.p1points and self.p2points < 4):
            if (self.p2points==2):
                P2res="trente"
            if (self.p2points==3):
                P2res="quarante"
            if (self.p1points==1):
                P1res="quinze"
            if (self.p1points==2):
                P1res="trente"
            result = P1res + "-" + P2res

        if (self.p1points > self.p2points and self.p2points >= 3):
            result = "avantage " + self.player1Name

        if (self.p2points > self.p1points and self.p1points >= 3):
            result = "avantage " + self.player2Name

        if (self.p1points>=4 and self.p2points>=0 and (self.p1points-self.p2points)>=2):
            result = "jeu pour " + self.player1Name
        if (self.p2points>=4 and self.p1points>=0 and (self.p2points-self.p1points)>=2):
            result = "jeu pour " + self.player2Name
        return result